[lao_phonetic] Version 1.1 - converted from Lao Script with John Durdin

mcdurdin opened the pull request:
This keyboard has been imported from a binary version in Lao Script for Windows by myself and my father @jmdurdin. The intention is to cleanup the .kmn in a future update and restore symbol names and comments (and cleanup the compile optimisations), but this version works correctly and has been requested by some users.

LornaSIL commented:
Could you consider putting the fonts in ```\release\shared\fonts\laoo\saysettha``` instead of the local ```assets``` folder?

LornaSIL commented:
Copyright date should be changed from "2015" to "2015-2018" in .kmn, readme.htm, welcome.htm (and subsequently hopefully the .kps and .kpj copyrights will be corrected.

I'm getting warning messages when I try to compile the .kmn file such as:
```lao_phonetic.kmn: Warning: line 2612  warning 209A: The rule will never be matched because its key code is never fired.```
That one is regarding this rule:
```   + "ñ" > "ຍ"  c line(0)```
but there are about 20 warning messages.

Also, in testing against the welcome.htm documentation:
```ko``` is producing ເກາະ and not ໂກະ (as documented) - actually I think the output of ko and ko` is reversed to what it should be.

Otherwise the keyboard seems to work properly, but because of the warnings, the .kmp file is not being created.

darcywong00 commented on release/l/lao_phonetic/source/lao_phonetic.kps:
```diff
@@ -0,0 +1,104 @@
+      <Name>Lao Phonetic</Name>
+      <ID>lao_phonetic</ID>
+      <Version>1.0</Version>
+      <Languages>
```
This might only pertain to touch keyboards, but is it good practice to set the keyboard font and display font to saysettha?

darcywong00 replied:
I just noticed lao_phonetic.js is included in the package, so the fonts will need to be set.

mcdurdin replied:
The lao_phonetic.js keyboard is desktop only; it shouldn't be in the package (I will remove it).

darcywong00 commented on release/l/lao_phonetic/lao_phonetic.kpj:
```diff
@@ -0,0 +1,129 @@
+<KeymanDeveloperProject>
+  <Options>
+    <BuildPath>$PROJECTPATH\build</BuildPath>
+    <CompilerWarningsAsErrors>False</CompilerWarningsAsErrors>
```
Is this intentional because of the multiple compiler warnings?

`This rule will never be matched because its key code is never fired.`

mcdurdin replied:
Yes.

jmdurdin commented:
It probably needs to be in assets to enable it to work with the touch keyboard – Marc should be able to confirm this.  Note that I have now found the original keyboard source files so there will probably be an update as soon as Marc can find the time to rebuild the package.

 

From: Lorna Evans [mailto:notifications@github.com] 
Sent: Tuesday, 13 November 2018 7:02 AM
To: keymanapp/keyboards <keyboards@noreply.github.com>
Cc: jmdurdin <jmdurdin@tavultesoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [keymanapp/keyboards] [lao_phonetic] Version 1.1 - converted from Lao Script with John Durdin (#533)

 

Could you consider putting the fonts in \release\shared\fonts\laoo\saysettha instead of the local assets folder?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/keymanapp/keyboards/pull/533#issuecomment-438010593> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AEzVTbKzfeY3_oHCEgrNszM8F3XeCa4cks5uudPCgaJpZM4YZVe6> .  <https://github.com/notifications/beacon/AEzVTcfTV76NG8eY5VV_KG0hc2jxEohTks5uudPCgaJpZM4YZVe6.gif>

mcdurdin commented:
Okay, I've updated the keyboard with the source that @jmdurdin found, which is much better. The `ñ` key will never be matched as the keyboard is setup as positional, but kmcomp does not pick up this issue when the key is in a store (`+ any(C) > index(CLao,1)`, line 237). Now this could be considered a bug or a gap in kmcomp...

However, in this case the `ñ` is being used as a placeholder: there is no single key which generates the `ຍ` consonant, but we still want the consonant in the `CLao` store. Changing this would require a significant number of changes to the keyboard for little benefit. There is no harm done.

If I [update kmcomp to address the missing warning](https://github.com/keymanapp/keyman/issues/1328), I will downgrade the warning to a hint. This would then allow the keyboard to compile even with 'treat warnings as errors' set (so no net change in compile behaviour for any keyboard already in the repo).

I have updated copyright in

darcywong00 commented on release/l/lao_phonetic/source/lao_phonetic.kps:
```diff
@@ -0,0 +1,98 @@
+    <Keyboard>
+      <Name>Lao Phonetic</Name>
+      <ID>lao_phonetic</ID>
+      <Version>1.0</Version>
```
Keyboard version is now 1.1

mcdurdin replied:
Fixed.
